variants/BLACK_F407VE/variant.h

```diff
@@ -0,0 +1,238 @@
+extern "C"{
+#endif // __cplusplus
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+extern const PinName digitalPin[];
+
+// Enum defining pin names to match digital pin number --> Dx
+// !!!
+// !!! Copy the digitalPin[] array in variant.cpp
+// !!! and remove all '_': PX_n --> PXn
+// !!! For NC, suffix by _x where x is the number of NC:
+// !!!   NC_1, NC_2,...
+// !!! For duplicated pin name, suffix by _x where x is the number of pin:
+// !!! PA7, PA7_2, PA7_3,...
+enum {
+/* Left Side 
+  Ext		Int     */
+  //5V		//5V
+  //5V		//5V
+  //3V3		//3V3
+  //3V3		//3V3
+  //GND		//GND
+  PE2,		PE3, 
+  PE4,		PE5, 	// PE4: BUT K0               PE5: BUT K1
+  PE6,		PC13, 
+  PC0,		PC1, 
+  PC2,		PC3, 
+  //VREF-	//VREF+ 
+  PA0,		PA1, 	// PA0(WK_UP): BUT K_UP
+  PA2,		PA3, 
+  PA4,		PA5, 
+  PA6,		PA7, 	// PA6: LED D2               PA7: LED D3  (active LOW)
+  PC4,		PC5, 
+  PB0,		PB1, 
+  PE7,		PE8, 
+  PE9,		PE10, 
+  PE11,		PE12, 
+  PE13,		PE14, 
+  PE15,		PB10, 
+  PB11,		PB12, 
+  PB13,		PB14, 
+/* Right Side 
+  Int		Ext     */
+  //3V3		//3V3
+  //3V3		//3V3
+  //BOOT0	//BOOT1
+  //GND		//GND
+  //GND		//GND
+  PE1,		PE0, 
+  PB9,		PB8, 
+  PB7,		PB6, 
+  PB5,		PB3, 
+  PD7,		PD6, 
+  PD5,		PD4, 
+  PD3,		PD2, 
+  PD1,		PD0, 
+  PC12,		PC11, 
+  PC10,		PA15, 
+  PA12,		PA11, // PA11: USB_DM               PA12: USB_DP
+  PA10,		PA9, 
+  PA8,		PC9, 
+  PC8,		PC7, 
+  PC6,		PD15, 
+  PD14,		PD13, 
+  PD12,		PD11, 
+  PD10,		PD9, 
+  PD8,		PB15, 
+  PB4,
+  // Analog pins
+  PA0_A,
+  PA1_A,
+  PA2_A,
+  PA3_A,
+  PA4_A,
+  PA5_A,
+  PB0_A,
+  PB1_A,
+  PC0_A,
+  PC1_A,
+  PC2_A,
+  PC3_A,
+  PC4_A,
+  PC5_A,
+  PEND
+};
+
+// This must be a literal with the same value as PEND
+// It is used with preprocessor tests (e.g. #if NUM_DIGITAL_PINS > 3)
+// so an enum will not work.
+#define NUM_DIGITAL_PINS        89
+
+// Allow to define Arduino style alias for analog input pin number --> Ax
+// All pins are digital, analog inputs are a subset of digital pins
+// and must be contiguous to be able to loop on each value
+// This must be a literal with a value less than or equal to MAX_ANALOG_INPUTS
+// defined in pin_arduino.h
+// It is used with preprocessor tests (e.g. #if NUM_ANALOG_INPUTS > 3)
+// so an enum will not work.
+// !!!
+// !!! It must be aligned with the number of analog PinName
+// !!! defined in digitalPin[] array in variant.cpp
+// !!!
+#define NUM_ANALOG_INPUTS       14
+// Define digital pin number of the first analog input  (i.e. which digital pin is A0)
+// First analog pin value (A0) must be greater than or equal to NUM_ANALOG_INPUTS
+#define NUM_ANALOG_FIRST        75
+
+// Below ADC, DAC and PWM definitions already done in the core
+// Could be redefined here if needed
+// ADC resolution is 12bits
+//#define ADC_RESOLUTION          12
+//#define DACC_RESOLUTION         12
+
+// PWM resolution
```

Comment:
ditto

Reply:
I would keep commented defines for future use..